### PR TITLE
Transformation cage

### LINKED ISSUE #491: Turn the Select Tool's bounding box into a transformation cage
Currently the Select Tool displays a rectangle around the bounds of the selected layers in the viewport. We should add squares (or ideally, triangles that point locally up) to the centers of the rectangle's edges as well as the corners which can be used for dragging to transform the selection. The cursor should become an up/down or left/right arrow when dragging anywhere along the edges of the rectangle. It should become a diagonal cursor when dragging near the corners of the rectangle.

And when the cursor is a bit further away from the edges and corners, but still within a reasonable proximity, it should display a rotational cursor (CSS doesn't provide one but we'll make one, you can start out by using the horizontal/vertical/diagonal ones parallel to the direction of rotational "torque"). Dragging when such a cursor is shown should cause rotation around the origin. (We'll add the ability to drag the origin point in a future issue.)

One complication: when the canvas is rotated in the viewport, we currently draw the bounding rectangle aligned upright in viewport space, which means it rotates in document space. This transformation is arguably a document space action which means it might be nice to display it in document space. One approach is to make the bounding box work in document space, the other option is to have it work in viewport space like it is now but transformation would then be applied from document to viewport space. Ideally, both are implemented and we figure out which is nicer and either make it a user preference or a toggle in the Select Tool's options bar.

### PR DESCRIPTION
**Pull request by 0HyperCube**
Closes #491

### CODE REVIEW

**Comment by Keavon** [comment]
It looks like this branch's [build](https://30d5fcc8.graphite-master.pages.dev/) also has the right side of the options bar missing. @mfish33 do you have any ideas why multiple Cloudflare builds now would have caused this randomly, unrelated to the code itself, and not in editor.graphite.design or localhost (yet?)?

**Comment by 0HyperCube** [comment]
@Keavon My best guess is that it is something to do with compiling the wasm / js with production optimisations. Can you try localhost with optimised wasm compiling?

**Comment by 0HyperCube** [comment]
Also @Keavon are there any more features that you would like in this before merging?

**Comment by Keavon** [comment]
> Also @Keavon are there any more features that you would like in this before merging?

I think the current set of features is a good checkpoint for the MVP implementation. We'll have more stuff like a draggable origin point and an options bar checkbox for local or global transformation (this is currently global where the bounding box is always viewport-relative not layer-relative). But those can all probably wait till after the alpha release. Current priority is getting the items on the sprint board done, like fonts and gradients. Let me code review this real quick then I think we can go ahead and merge.

File: editor/src/viewport_tools/tools/select.rs
Comment (Keavon) [inline editor/src/viewport_tools/tools/select.rs:319]: Is there any other clean way we could do this that doesn't involve "returning data" to the arguments of the caller? This feels too much like a C hack which Rust's superior feature set should help us avoid. If you think this is the cleanest way, no worries.

File: editor/src/viewport_tools/tools/select.rs
Comment (Keavon) [inline editor/src/viewport_tools/tools/select.rs:420]: Typo: `origional` -> `original`

File: editor/src/viewport_tools/tools/select.rs
Comment (Keavon) [inline editor/src/viewport_tools/tools/select.rs:499]: This is a pretty cool use of Rust's match statement.

**Comment by Keavon** [comment]
> Also @Keavon are there any more features that you would like in this before merging?

Oh yeah, that one other feature request about removing the corner handles if the width or height of the layer (in viewport pixels) is too small to fit the three handles. Although if you want to do that in a followup PR, that's fine too.

File: editor/src/viewport_tools/tools/select.rs
Comment (Keavon) [inline editor/src/viewport_tools/tools/select.rs:707]: Nitpick: using the full word `vector` would help avoid confusion with Rust's `Vec` data type.

**Review by Keavon — APPROVED** [review]
Good to merge after resolving comments! The editor is becoming more and more like a real graphics suite every day largely thanks to your awesome work.

File: editor/src/viewport_tools/tools/select.rs
Comment (0HyperCube) [reply editor/src/viewport_tools/tools/select.rs:319]: Also similar to C# out parameters :). This probably should have been refactored when I moved the pivot out of the SelectedEdges struct.

File: editor/src/viewport_tools/tools/select.rs
Comment (0HyperCube) [reply editor/src/viewport_tools/tools/select.rs]: I think end_offset is probably a better name for this. I have renamed all the other occurrences of this pattern too.

Comment (Keavon) [reply editor/src/viewport_tools/tools/select.rs]: Perfect, even better.

**Pull request merged**
